storage/integration_test.go

```diff
@@ -2671,8 +2670,15 @@ func TestIntegration_DeleteObjectInBucketWithRetentionPolicy(t *testing.T) {
 	}
 
 	// Remove the retention period
-	h.mustUpdateBucket(bkt, BucketAttrsToUpdate{RetentionPolicy: &RetentionPolicy{RetentionPeriod: 0}})
-	h.mustDeleteObject(oh)
+	h.mustUpdateBucket(bkt, BucketAttrsToUpdate{RetentionPolicy: &RetentionPolicy{}})
+	// Deleting with retry, as bucket metadata changes
+	// can take some time to propagate.
+	err := retry(ctx, func() error {
+		return oh.Delete(ctx)
+	}, nil)
+	if err != nil {
+		h.t.Fatalf("%s: object delete: %v", loc(), err)
+	}
```

Comment:
I've been running tests for a couple of days. Seems like it usually takes 2-4 sec to propagate metadata changes, but sometime it even goes to 8 sec. Adding `retry()` as it retries the call for 11 sec with 0.2 sec delay. After adding it I didn't get any single fail locally.

Reply:
Nice, thanks for the details on how you tested locally-- very helpful.